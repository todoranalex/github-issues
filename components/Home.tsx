import {useNavigation, useTheme} from '@react-navigation/native';
import React, {FunctionComponent, useState} from 'react';
import {
  Alert,
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  useWindowDimensions,
  View,
} from 'react-native';
import GithubIcon from 'react-native-vector-icons/Octicons';

export default () => {
  const theme = useTheme();
  const navigation = useNavigation();
  const {height, width} = useWindowDimensions();
  const [githubDetails, setGithubDetails] = useState<{
    repository: string;
    organization: string;
  }>({
    repository: '',
    organization: '',
  });

  return (
    // <KeyboardAvoidingView style={{flex: 1}} behavior={'padding'}>
    // <ScrollView>
    <View style={styles.container}>
      <GithubIcon name="mark-github" color={theme.colors.primary} size={128} />
      <Text style={{color: theme.colors.primary, ...styles.subtitle}}>
        Your GitHub issues fetcher app
      </Text>
      <View style={{width: width - 128}}>
        <Input
          label={'Organization'}
          value={githubDetails.organization}
          placeholder={'Insert organization here'}
          onChangeText={organization => {
            setGithubDetails(state => {
              return {...state, organization};
            });
          }}
        />
        <Input
          label={'Repository'}
          value={githubDetails.repository}
          placeholder={'Insert repository here'}
          onChangeText={repository => {
            setGithubDetails(state => {
              return {...state, repository};
            });
          }}
        />

        <Button
          text={'Fetch Issues'}
          icon={'rocket'}
          onPress={() => {
            const {repository, organization} = githubDetails;
            if (repository.length > 0 && organization.length > 0) {
              navigation.navigate('Issues', {
                organization: organization.toLowerCase(),
                repository: repository.toLowerCase(),
              });
            } else {
              Alert.alert(
                'Error',
                'Please input an organization and a repository',
              );
            }
          }}
        />
        <Button
          text={'Bookmarks'}
          icon={'bookmark'}
          onPress={() => {
            navigation.navigate('Bookmarks', {});
          }}
        />
      </View>
    </View>
    // </ScrollView>
    // </KeyboardAvoidingView>
  );
};

const Input: FunctionComponent<{
  label: string;
  value: string;
  placeholder: string;
  onChangeText(value: string): void;
}> = ({label, value, placeholder, onChangeText}): JSX.Element => {
  const theme = useTheme();
  return (
    <React.Fragment>
      <Text style={{color: theme.colors.primary, ...styles.textInputInfo}}>
        {label}
      </Text>
      <TextInput
        onChangeText={text => {
          onChangeText(text);
        }}
        keyboardAppearance="dark"
        style={{
          ...styles.textInput,
          borderColor: theme.colors.primary,
          color: theme.colors.primary,
        }}
        placeholderTextColor={theme.colors.text}
        placeholder={placeholder}
        value={value}
      />
    </React.Fragment>
  );
};

export const Button: FunctionComponent<{
  text: string;
  icon?: string;
  borderColor?: string;
  onPress(): void;
}> = ({text, icon, borderColor, onPress}) => {
  const theme = useTheme();
  return (
    <TouchableOpacity
      onPress={onPress}
      style={{
        ...styles.button,
        borderColor: borderColor ?? theme.colors.primary,
      }}>
      <View style={styles.buttonInnerContainer}>
        <Text style={{...styles.buttonText, color: theme.colors.primary}}>
          {text}
        </Text>
        {icon && (
          <GithubIcon name={icon} size={24} color={theme.colors.primary} />
        )}
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  subtitle: {
    marginTop: 24,
    marginBottom: 16,
  },
  textInputInfo: {
    marginBottom: 8,
  },
  textInput: {
    borderWidth: 1,
    height: 48,
    marginBottom: 16,
    paddingLeft: 8,
  },
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    borderWidth: 1,
    height: 48,
    marginBottom: 8,
  },
  buttonInnerContainer: {flexDirection: 'row', paddingHorizontal: 48},
  buttonText: {
    fontSize: 20,
    marginRight: 8,
  },
});
