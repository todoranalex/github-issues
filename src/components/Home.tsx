import {useNavigation, useTheme} from '@react-navigation/native';
import React, {useState} from 'react';
import {
  Alert,
  ScrollView,
  StyleSheet,
  Text,
  useWindowDimensions,
  View,
} from 'react-native';
import GithubIcon from 'react-native-vector-icons/Octicons';
import {Button, Input} from './IssuesComponents';

/***
 * Component used as the main entry point for the app.
 */
export default () => {
  const theme = useTheme();
  const navigation = useNavigation();
  const {width} = useWindowDimensions();
  const [githubDetails, setGithubDetails] = useState<{
    repository: string;
    organization: string;
  }>({
    repository: '',
    organization: '',
  });

  return (
    <ScrollView contentContainerStyle={styles.container}>
      <GithubIcon name="mark-github" color={theme.colors.primary} size={128} />
      <Text style={{color: theme.colors.primary, ...styles.subtitle}}>
        Your GitHub issues fetcher app
      </Text>
      <View style={{width: width - 128}}>
        <Input
          testID={'orgInputTestID'}
          label={'Organization'}
          value={githubDetails.organization}
          placeholder={'e.g facebook'}
          onChangeText={organization => {
            setGithubDetails(state => {
              return {...state, organization};
            });
          }}
        />
        <Input
          testID={'repoInputTestID'}
          label={'Repository'}
          value={githubDetails.repository}
          placeholder={'e.g react-native'}
          onChangeText={repository => {
            setGithubDetails(state => {
              return {...state, repository};
            });
          }}
        />
        <Button
          testID="fetchIssuesButtonTestID"
          text={'Fetch Issues'}
          icon={'rocket'}
          onPress={() => {
            const {repository, organization} = githubDetails;
            if (repository.length > 0 && organization.length > 0) {
              navigation.navigate('Issues', {
                organization: organization.toLowerCase(),
                repository: repository.toLowerCase(),
              });
            } else {
              Alert.alert(
                'Error',
                'Please input an organization and a repository',
              );
            }
          }}
        />
        <Button
          testID="bookmarksButtonTestID"
          text={'Bookmarks'}
          icon={'heart-fill'}
          onPress={() => {
            navigation.navigate('Bookmarks', {});
          }}
        />
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  subtitle: {
    marginTop: 24,
    marginBottom: 16,
  },
});
