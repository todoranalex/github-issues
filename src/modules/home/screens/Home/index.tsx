import {useTheme} from '@react-navigation/native';
import React from 'react';
import {
  ScrollView,
  StyleSheet,
  Text,
  useWindowDimensions,
  View,
} from 'react-native';
import GithubIcon from 'react-native-vector-icons/Octicons';
import {Input, Button} from '../../../generic/components';
import useHome from '../../hooks/useHome';

/***
 * Component used as the main entry point for the app.
 */
const Home = () => {
  const theme = useTheme();
  const {width} = useWindowDimensions();

  const {
    githubDetails,
    onOrganizationChange,
    onRepositoryChange,
    onFetchIssuesPressed,
    onBookmarksPressed,
  } = useHome();

  return (
    <ScrollView contentContainerStyle={styles.container}>
      <GithubIcon name="mark-github" color={theme.colors.primary} size={128} />
      <Text style={{color: theme.colors.primary, ...styles.subtitle}}>
        Your GitHub issues fetcher app
      </Text>
      <View style={{width: width - 128}}>
        <Input
          testID={'orgInputTestID'}
          label={'Organization'}
          value={githubDetails.organization}
          placeholder={'e.g facebook'}
          onChangeText={onOrganizationChange}
        />
        <Input
          testID={'repoInputTestID'}
          label={'Repository'}
          value={githubDetails.repository}
          placeholder={'e.g react-native'}
          onChangeText={onRepositoryChange}
        />
        <Button
          testID="fetchIssuesButtonTestID"
          text={'Fetch Issues'}
          icon={'rocket'}
          onPress={onFetchIssuesPressed}
        />
        <Button
          testID="bookmarksButtonTestID"
          text={'Bookmarks'}
          icon={'heart-fill'}
          onPress={onBookmarksPressed}
        />
      </View>
    </ScrollView>
  );
};

export default Home;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  subtitle: {
    marginTop: 24,
    marginBottom: 16,
  },
});
