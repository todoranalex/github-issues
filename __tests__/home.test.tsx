import React from 'react';
import {render} from '@testing-library/react-native';
import Home from '../src/components/Home';
import renderer from 'react-test-renderer';

describe('Home Components', () => {
  test('test org input mount', () => {
    const {getByTestId} = render(<Home />);
    expect(getByTestId('orgInputTestID')).toBeDefined();
  });
  test('test repo input mount', () => {
    const {getByTestId} = render(<Home />);
    expect(getByTestId('repoInputTestID')).toBeDefined();
  });
  test('test fetch issues button mount', () => {
    const {getByTestId} = render(<Home />);
    expect(getByTestId('fetchIssuesButtonTestID')).toBeDefined();
  });
  test('test bookmarks button  mount', () => {
    const {getByTestId} = render(<Home />);
    expect(getByTestId('bookmarksButtonTestID')).toBeDefined();
  });
  test('Home snapshot', () => {
    const snap = renderer.create(<Home />).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
