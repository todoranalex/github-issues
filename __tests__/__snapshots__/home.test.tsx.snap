// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Components Home snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "rgb(0, 122, 255)",
            "fontSize": 128,
          },
          undefined,
          Object {
            "fontFamily": "Octicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "rgb(0, 122, 255)",
          "marginBottom": 16,
          "marginTop": 24,
        }
      }
    >
      Your GitHub issues fetcher app
    </Text>
    <View
      style={
        Object {
          "width": 622,
        }
      }
    >
      <View
        testID="orgInputTestID"
      >
        <Text
          style={
            Object {
              "color": "rgb(0, 122, 255)",
              "marginBottom": 8,
            }
          }
        >
          Organization
        </Text>
        <TextInput
          keyboardAppearance="dark"
          onChangeText={[Function]}
          placeholder="e.g facebook"
          placeholderTextColor="rgb(28, 28, 30)"
          style={
            Object {
              "borderColor": "rgb(0, 122, 255)",
              "borderWidth": 1,
              "color": "rgb(0, 122, 255)",
              "height": 48,
              "marginBottom": 16,
              "paddingLeft": 8,
            }
          }
          value=""
        />
      </View>
      <View
        testID="repoInputTestID"
      >
        <Text
          style={
            Object {
              "color": "rgb(0, 122, 255)",
              "marginBottom": 8,
            }
          }
        >
          Repository
        </Text>
        <TextInput
          keyboardAppearance="dark"
          onChangeText={[Function]}
          placeholder="e.g react-native"
          placeholderTextColor="rgb(28, 28, 30)"
          style={
            Object {
              "borderColor": "rgb(0, 122, 255)",
              "borderWidth": 1,
              "color": "rgb(0, 122, 255)",
              "height": 48,
              "marginBottom": 16,
              "paddingLeft": 8,
            }
          }
          value=""
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "rgb(0, 122, 255)",
            "borderWidth": 1,
            "height": 48,
            "justifyContent": "center",
            "marginBottom": 8,
            "opacity": 1,
          }
        }
        testID="fetchIssuesButtonTestID"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 48,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(0, 122, 255)",
                "fontSize": 20,
                "marginRight": 8,
              }
            }
          >
            Fetch Issues
          </Text>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "rgb(0, 122, 255)",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "Octicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "rgb(0, 122, 255)",
            "borderWidth": 1,
            "height": 48,
            "justifyContent": "center",
            "marginBottom": 8,
            "opacity": 1,
          }
        }
        testID="bookmarksButtonTestID"
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "paddingHorizontal": 48,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(0, 122, 255)",
                "fontSize": 20,
                "marginRight": 8,
              }
            }
          >
            Bookmarks
          </Text>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "rgb(0, 122, 255)",
                  "fontSize": 24,
                },
                undefined,
                Object {
                  "fontFamily": "Octicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
